### ISSUE TITLE
Apple Mach-O Linker Error

### ISSUE THREAD

**Issue by CrazyGoose** [dropbox/dropbox-sdk-obj-c#146]
Pod is installed correctly.  Followed all setup as outlined but I am getting the following errors:

  "_OBJC_CLASS_$_DBClientsManager", referenced from:

clang: error: linker command failed with exit code 1 (use -v to see invocation)

If I comment out: [DBClientsManager setupWithAppKey:@"XXXXXXXXXX"]; everything complies without issue.

**Comment by scobbe** [comment]
@CrazyGoose: Do you have the `.xcworkspace` file open and **not** the `.xcodeproj` file?

**Comment by CrazyGoose** [comment]
Yes, I am using xcworkspace.

**Comment by scobbe** [comment]
@CrazyGoose: To get the obvious stuff out of the way, would you mind confirming that you've restarted Xcode, and then run `cmd+option+shift+k` and done a clean build?

**Comment by CrazyGoose** [comment]
Yes,

I have cleaned derived data, restarted xcode, uninstalled pod, reinstalled pod.... many combination of each.

It is the minute I put in the code: [DBClientsManager setupWithAppKey:@"XXXXXXXXXX"];
The actual line of code does not show any errors. But it just wont compile.

**Comment by scobbe** [comment]
Would you mind trying the step [here](https://github.com/dropbox/dropbox-sdk-obj-c#cocoapods) that involves adding `$(inherited)` to your project's **Other Linker Flags** in Build Settings, and ensure that the `-ObjC flag` is included in **Other Linker Flags**?

`DBClientsManager` uses categories, which can fail at link time sometimes if the `-ObjC` flag isn't added

**Comment by CrazyGoose** [comment]
![screen shot 2017-05-05 at 5 21 20 pm](https://cloud.githubusercontent.com/assets/3039813/25764707/48489800-31b7-11e7-8e5d-5fcb7b85ccab.png)

**Comment by scobbe** [comment]
@CrazyGoose: Ok, thanks. Another thing – set `BUILD ACTIVE ARCHITECTURE ONLY` to `NO` for both debug and release in your project's build settings. Let me know if that helps at all.

**Comment by scobbe** [comment]
Also, if you run `pod --version` and it's less than 0.25.0, you may need to update (`sudo gem update cocoapods`)

**Comment by CrazyGoose** [comment]
Pod version 1.2.1
Changed Active Arch.. to NO for Debug.  Release was already NO.

**Comment by scobbe** [comment]
@CrazyGoose: Sorry for the radio silence here. Wanted to follow up and see if this issue was still happening for you. If it is, would you mind sending in a redacted Xcode project to the following link:

https://www.dropbox.com/developers/contact

That way, we can debug it on our end and get to the bottom of this. Either way, let me know. Thank you!

**Comment by Schwiiz** [comment]
Hi scobbe, could you please help me with the same problem: 
ld: framework not found Pods-DBRoulette
clang: error: linker command failed with exit code 1 (use -v to see invocation)
I'm trying to run the sample code.. 
thanks!

**Comment by scobbe** [comment]
@Schwiiz: I assume you tried everything above?

Based on your error, it seems like you don't have the `.xcworkspace` file open (but instead the `.xcodeproj` file).

Would you mind confirming that you have the `.xcworkspace` file open while attempting to build?

**Comment by Schwiiz** [comment]
unfortunately, it's .xcworkspace. it looks like this:

![screen shot 2017-06-05 at 18 58 10](https://cloud.githubusercontent.com/assets/13655824/26794185/ff2e5dd0-4a20-11e7-9b7e-a79c7607f968.png)

**Comment by scobbe** [comment]
@Schwiiz: Oh this is the example project. Did you run `pod install` in the directory that contains the `Podfile`?

We don't distribute the framework code (ObjectiveDropboxOfficial) in the examples directly, but instead require you to fetch them. Let me know if this works.

**Comment by scobbe** [comment]
Oh wait, yeah we don't distribute the workspace files either, so you must have...

**Comment by scobbe** [comment]
@Schwiiz: Would you mind trying this:

In Project Navigator in the folder Pods look for Pods_DBRoulette.framework which is red in the above screenshot. It should also be present in Linked Frameworks and Libraries. Remove both references and see if that helps.

**Comment by Schwiiz** [comment]
@scobbe ok, let me try. thanks.

**Comment by scobbe** [comment]
@Schwiiz: Any word?

**Comment by scobbe** [comment]
Ping @Schwiiz and @CrazyGoose.

Any update? Where you able to get this to work?

**Comment by igiso** [comment]
I get a mac o linker Error 
_OBJC_CLASS_$_DBClientsManager", referenced from: objc-class-ref in ofxDropBox.o

When I run on Ipad,
but on simulator everything runs ok?
Any Ideas?

<img width="311" alt="screen shot 2017-10-10 at 15 36 48" src="https://user-images.githubusercontent.com/2058400/31386901-25b713e8-add1-11e7-9313-ee6a6b0434ef.png">
<img width="1957" alt="screen shot 2017-10-10 at 15 36 25" src="https://user-images.githubusercontent.com/2058400/31386902-25f4daf2-add1-11e7-88f0-6a29fe9fb683.png">
<img width="818" alt="screen shot 2017-10-10 at 15 35 59" src="https://user-images.githubusercontent.com/2058400/31386903-263f4628-add1-11e7-87af-f069362fe3ce.png">
<img width="833" alt="screen shot 2017-10-10 at 15 28 09" src="https://user-images.githubusercontent.com/2058400/31386904-266701c2-add1-11e7-9d6b-cf19e3bafb16.png">

**Comment by greg-db** [comment]
@igiso Have you already tried the troubleshooting steps discussed earlier in this thread?

**Comment by igiso** [comment]
Yes I've tried, everything, I need to clarify, that my project works on Simulator, the linking problem is when testing on Device.

**Comment by igiso** [comment]
ps
I didn't run those Pod files, I just dragged the ObjectiveDropboxOfficial.framework into the project
 
I tried to run them,  but it made any difference

<img width="572" alt="screen shot 2017-10-19 at 16 16 52" src="https://user-images.githubusercontent.com/2058400/31772649-104bbbca-b4e9-11e7-99af-58b4415a0150.png">

**Comment by igiso** [comment]
On the simulator I get this warning but it compiles:

<img width="754" alt="screen shot 2017-08-30 at 17 46 51" src="https://user-images.githubusercontent.com/2058400/31772862-90b9943a-b4e9-11e7-9bcc-f53566dda060.png">

**Comment by igiso** [comment]
I am putting a reference to this I think it is related:

https://github.com/drewmccormack/ensembles/issues/208

**Comment by igiso** [comment]
I've solved the problem, I had to rebuild the framework for the particular device and then relink it,
but now it doesn't work on the simulator,

I created two projects with shared code but different linked frameworks..

**Comment by igiso** [comment]
now the other issue I have is that I get redirected to the dropbox but I do not continue , it is stuck in an authentication loop I click allow on dropbox but it keeps asking for authentication .. I must have skiped a part in the code..

**Comment by igiso** [comment]
so apparently I must enable Application does not run in background to NO
for the code to work..

IS there a way to disable this? I want the app to shut down each time it runs..

**Comment by greg-db** [comment]
@igiso The SDK should work both on device and in the simulator. You'll need to pick one of the supported installation methods though, and follow all of the documented steps:

https://github.com/dropbox/dropbox-sdk-obj-c#sdk-distribution

It sounds like you only manually dragged in the ObjectiveDropboxOfficial.framework. To get this working, please try again using one of the officially supported installation methods.

**Comment by monish152** [comment]
![screen shot 2019-01-13 at 11 58 52 pm](https://user-images.githubusercontent.com/1301491/51089240-077ef100-1790-11e9-91ed-be95ace80f2e.png)
@scobbe getting same issue while adding manually. There is only xcodeproj file
